metrics/metrics.go

```diff
@@ -0,0 +1,53 @@
+package metrics
+
+import (
+	"context"
+	"errors"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var Enabled = false
```

Comment:
I think we make this private by setting it in the `New()`, having both bool flag and this public global variable seems redundant to me.

Reply:
Will fix in another issue.